core/fetcher/src/main/kotlin/fetcher/Fetcher.kt

```diff
@@ -0,0 +1,41 @@
+package fetcher
+
+import network.NetworkModule
+import network.NetworkModuleFactory
+import network.SourceNotFoundException
+import fetcher.rss.RssParser
+import fetcher.rss.RssParserFactory
+
+interface Fetcher {
+    @Throws(FeedCouldNotBeObtainedException::class)
+    suspend fun obtainFeed(url: String): Feed
+}
+
+object FetcherFactory {
+    fun create(): Fetcher {
```

Comment:
Can we flatten these slightly via just having a top-level `createFetcher()`?

Reply:
I'd rather introduce some DI library and get rid of them altogether.

But if you like it more I'll do that

Reply:
We can make a decision on a DI library in a later PR :)